### PR TITLE
[IMP] l10n_it_edi: Add italian invoicing fields to invoice PDF

### PR DESCRIPTION
**Pull request by seifgneedy**
Some fields related to Italian invoicing are included in XML but missing from the invoice PDF. Added these fields to report_invoice for users to have more visibility for what's included in the XML submitted to Sdl.

Task [link](https://www.odoo.com/odoo/project.task/3794551)
task-3794551

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/views/report_invoice.xml [hunk 1]
```diff
@@ -256,13 +256,11 @@
...
-                                    <p name="payment_communication">
-                                        Payment Communication: <span class="fw-bold" t-field="o.payment_reference">INV/2023/00001</span>
-                                        <t t-if="o.partner_bank_id">
-                                            <br/> on this account: <span t-field="o.partner_bank_id" class="fw-bold"/>
-                                        </t>
-                                    </p>
+                                <div class="mb-3" t-if="o.move_type in ('out_invoice', 'in_refund') and o.payment_reference"  name="payment_communication">
+                                    Payment Communication: <span class="fw-bold" t-field="o.payment_reference">INV/2023/00001</span>
+                                    <t t-if="o.partner_bank_id">
+                                        <br/> on this account: <span t-field="o.partner_bank_id" class="fw-bold"/>
```
Comment (smetl) [inline addons/account/views/report_invoice.xml:262]: what is the reason behind this change in stable version? `<p/>` is named (`payment_communication`) so probably used in existing xpaths

File: addons/l10n_it_edi/views/report_invoice.xml [hunk 2]
```diff
@@ -2,12 +2,19 @@
...
-                <t t-if="o.l10n_it_cup"><b>CUP: </b><span t-field="o.l10n_it_cup"/><br/></t>
-            </div>
+                <t t-if="o.l10n_it_stamp_duty"><b>Stamp Duty: </b><span t-field="o.l10n_it_stamp_duty"></span></t><br/>
+                <t t-if="o.l10n_it_ddt_id"><b>Transport Document: </b><span t-field="o.l10n_it_ddt_id"></span></t><br/>
+                <div t-if="o.l10n_it_origin_document_type" name="pa_fields">
+                    <b><span t-field="o.l10n_it_origin_document_type"/></b>: <span t-field="o.l10n_it_origin_document_name"/><br/>
+                    <t t-if="o.l10n_it_origin_document_date"><b>Document Date: </b><span t-field="o.l10n_it_origin_document_date"/><br/></t>
+                    <t t-if="o.l10n_it_cig"><b>CIG: </b><span t-field="o.l10n_it_cig"/><br/></t>
+                    <t t-if="o.l10n_it_cup"><b>CUP: </b><span t-field="o.l10n_it_cup"/><br/></t>
+                </div>
```
Comment (smetl) [inline addons/l10n_it_edi/views/report_invoice.xml:12]: weird indentation

File: addons/l10n_it_edi/views/report_invoice.xml [hunk 3]
```diff
@@ -2,12 +2,19 @@
 <odoo>
     <template id="report_invoice_document" inherit_id="account.report_invoice_document">
         <div name="comment" position="before">
-            <div t-if="o.l10n_it_origin_document_type" name="pa_fields">
-                <b><span t-field="o.l10n_it_origin_document_type"/></b>: <span t-field="o.l10n_it_origin_document_name"/><br/>
-                <t t-if="o.l10n_it_origin_document_date"><b>Document Date: </b><span t-field="o.l10n_it_origin_document_date"/><br/></t>
-                <t t-if="o.l10n_it_cig"><b>CIG: </b><span t-field="o.l10n_it_cig"/><br/></t>
-                <t t-if="o.l10n_it_cup"><b>CUP: </b><span t-field="o.l10n_it_cup"/><br/></t>
-            </div>
+                <t t-if="o.l10n_it_stamp_duty"><b>Stamp Duty: </b><span t-field="o.l10n_it_stamp_duty"></span></t><br/>
```
Comment (smetl) [inline addons/l10n_it_edi/views/report_invoice.xml:5]: `<br/>` is not included in the t-if condition. Are you not addind extra empty line in case of missing data?

File: addons/l10n_it_edi/views/report_invoice.xml [hunk 4]
```diff
@@ -2,12 +2,19 @@
...
+                <div t-if="o.l10n_it_origin_document_type" name="pa_fields">
+                    <b><span t-field="o.l10n_it_origin_document_type"/></b>: <span t-field="o.l10n_it_origin_document_name"/><br/>
+                    <t t-if="o.l10n_it_origin_document_date"><b>Document Date: </b><span t-field="o.l10n_it_origin_document_date"/><br/></t>
+                    <t t-if="o.l10n_it_cig"><b>CIG: </b><span t-field="o.l10n_it_cig"/><br/></t>
+                    <t t-if="o.l10n_it_cup"><b>CUP: </b><span t-field="o.l10n_it_cup"/><br/></t>
+                </div>
+        </div>
+
+        <div name="payment_communication" position="inside">
+            <t t-set="payments" t-value="o.line_ids.filtered(lambda line: line.account_id.account_type in ('asset_receivable', 'liability_payable'))"/>
```
Comment (smetl) [inline addons/l10n_it_edi/views/report_invoice.xml:16]: Comment 1)
```suggestion
            <t t-set="payments" t-value="o.line_ids.filtered(lambda line: line.display_type == 'payment_term')"/>
```
Comment 2)
`payments` is confusing here. At least, rename it to `term_lines`

File: addons/account/views/report_invoice.xml (hunk 1 above)
Comment (seifgneedy) [reply addons/account/views/report_invoice.xml]: My bad, I was misunderstanding how template works :)

File: addons/l10n_it_edi/views/report_invoice.xml [hunk 5]
```diff
@@ -2,12 +2,19 @@
...
                 <t t-if="o.l10n_it_origin_document_date"><b>Document Date: </b><span t-field="o.l10n_it_origin_document_date"/><br/></t>
                 <t t-if="o.l10n_it_cig"><b>CIG: </b><span t-field="o.l10n_it_cig"/><br/></t>
                 <t t-if="o.l10n_it_cup"><b>CUP: </b><span t-field="o.l10n_it_cup"/><br/></t>
             </div>
         </div>
+
+        <p name="payment_communication" position="inside">
+            <t t-set="term_lines" t-value="o.line_ids.filtered(lambda line: line.display_type == 'payment_term')"/>
+            <br/><b>Payment Conditions:</b> <t t-if="len(term_lines) == 1">TP02 pagamento completo</t><t t-else="">TP01 pagamento a rate</t><br/>
+        </p>
```
Comment (smetl) [inline addons/l10n_it_edi/views/report_invoice.xml:18]: The template is inherited for everyone, not only the Italian companies. It means, stuff like
`<t t-if="o.l10n_it_stamp_duty"><b>Stamp Duty: </b><span t-field="o.l10n_it_stamp_duty"></span><br/></t>`
is working because `l10n_it_stamp_duty` will not be set for not italian companies. However, that's not the case of this block. You need at least one `t-if` ensuring this block will only be added for IT.

**Review by smetl — CHANGES_REQUESTED** [review]

File: addons/l10n_it_edi/views/report_invoice.xml [hunk 6]
```diff
@@ -2,12 +2,21 @@
...
             <div t-if="o.l10n_it_origin_document_type" name="pa_fields">
                 <b><span t-field="o.l10n_it_origin_document_type"/></b>: <span t-field="o.l10n_it_origin_document_name"/><br/>
                 <t t-if="o.l10n_it_origin_document_date"><b>Document Date: </b><span t-field="o.l10n_it_origin_document_date"/><br/></t>
                 <t t-if="o.l10n_it_cig"><b>CIG: </b><span t-field="o.l10n_it_cig"/><br/></t>
                 <t t-if="o.l10n_it_cup"><b>CUP: </b><span t-field="o.l10n_it_cup"/><br/></t>
             </div>
         </div>
+
+        <p name="payment_communication" position="inside">
+            <t t-if="o.company_id.country_id.code == 'IT'">
```
Comment (smetl) [inline addons/l10n_it_edi/views/report_invoice.xml:16]:
```suggestion
            <t t-if="o.country_code == 'IT'">
```
It prevents to wrongly use the country instead of the fiscal one. For reference:
```py
country_code = fields.Char(related='company_id.account_fiscal_country_id.code', readonly=True)
```

**Review by smetl — APPROVED** [review]
@robodoo r+
